syntax = "proto3";

package plotinus;

import "common.proto";

// 基座管理服务
service BaseService {
  // 获取基座列表
  rpc ListBases(ListBasesRequest) returns (ListBasesResponse);

  // 获取基座详情
  rpc GetBase(GetBaseRequest) returns (GetBaseResponse);

  // 创建基座
  rpc CreateBase(CreateBaseRequest) returns (CreateBaseResponse);

  // 更新基座
  rpc UpdateBase(UpdateBaseRequest) returns (UpdateBaseResponse);

  // 删除基座
  rpc DeleteBase(DeleteBaseRequest) returns (DeleteBaseResponse);
}

// 基座信息
message Base {
  int32 id = 1;
  string base_name = 2;
  string info = 3;
  int32 user_id = 4;
}

// 获取基座列表请求
message ListBasesRequest {
  string user_id = 1;
}

// 获取基座列表响应
message ListBasesResponse {
  Status status = 1;
  repeated Base bases = 2;
}

// 获取基座详情请求
message GetBaseRequest {
  string user_id = 1;
  int32 base_id = 2;
}

// 获取基座详情响应
message GetBaseResponse {
  Status status = 1;
  Base base = 2;
}

// 创建基座请求
message CreateBaseRequest {
  string user_id = 1;
  string base_name = 2;
  string info = 3;
}

// 创建基座响应
message CreateBaseResponse {
  Status status = 1;
  Base base = 2;
}

// 更新基座请求
message UpdateBaseRequest {
  string user_id = 1;
  int32 base_id = 2;
  string base_name = 3;
  string info = 4;
}

// 更新基座响应
message UpdateBaseResponse {
  Status status = 1;
  Base base = 2;
}

// 删除基座请求
message DeleteBaseRequest {
  string user_id = 1;
  int32 base_id = 2;
}

// 删除基座响应
message DeleteBaseResponse {
  Status status = 1;
}
